### PR TITLE
docs: add links to hpx recepies

### LINKED ISSUE #4016: Improving Documentation
Since I was asked in IRC to do so, here are three attempts of mine to document some basic recipes:

https://gitlab.com/snippets/1821389
https://gitlab.com/snippets/1822983
https://gitlab.com/snippets/1828131

Maybe they are helpful for people working on HPX Documentation.
Cheers
~Chris

### PR DESCRIPTION
**Pull request by bhumitattarde**
Fixes #4016

### CODE REVIEW

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

File: docs/sphinx/additional_material.rst [hunk 1]
```diff
@@ -17,3 +17,8 @@ Additional material
   * `Slides <https://github.com/STEllAR-GROUP/tutorials/tree/master/cscs2016>`_
 * `Tutorials repository <https://github.com/STEllAR-GROUP/tutorials>`_
 * `STE||AR Group blog posts <http://stellar-group.org/blog/>`_
+* Basic HPX recipes
```
Comment (msimberg) [inline docs/sphinx/additional_material.rst:20]:
```suggestion
* Basic |hpx| recipes
```

**Review by msimberg — CHANGES_REQUESTED** [review]
Thanks @bhumitattarde! Just a minor formatting request, otherwise this looks good.

File: docs/sphinx/additional_material.rst (hunk 1 above)
Comment (bhumitattarde) [reply docs/sphinx/additional_material.rst]: Done :)

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
bors merge

**Pull request merged**